crates/forge/src/multi_runner.rs

```diff
@@ -391,8 +395,24 @@ impl MultiContractRunnerBuilder {
                 deployable_contracts.insert(id.clone(), (abi.clone(), bytecode, libs_to_deploy));
             }
 
-            if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
-                known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+            if !use_zk {
+                if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
+                    known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+                }
+            }
+        }
+        if use_zk {
+            let zk_contracts =
+                zk_output.unwrap().with_stripped_file_prefixes(root).into_artifacts();
+            for (id, contract) in zk_contracts {
+                if let Some(abi) = contract.abi.as_ref() {
```

Comment:
If you need the abi for real, the contract there does not have it sometimes, the source of truth is in metadata, as a raw json. See: https://github.com/matter-labs/foundry-zksync/blob/56b536b728b2ca1c841343ac6d3964b93d05b97a/crates/forge/bin/cmd/create.rs#L203 for how do we do it for `forge create` (maybe extract a common method for it)

Reply:
It seems this isn't something that will be present for a long time (the abi extraction from metadata) I wouldn't create a method if that is the case.